### ISSUE TITLE
Missing `fullNameOverride` common helm chart functionality

### ISSUE THREAD

**Issue by gwvandesteeg** [neo4j-contrib/neo4j-helm#45]
Almost all commonly used helm charts have the ability to use a `fullNameOverride` value in the values.yaml which allows the end user to override the name of the deployed components to be able to deploy them without the helm Release value in their name (so you don't end up with a `neo4j-neo4j` name of the deployment when all you want is a `neo4j` entry).

Request:
- Expand the naming in the `_helpers.tpl` to be able to use this value.

**Comment by moxious** [comment]
Can you provide an example of a helm chart that does what you're requesting so I can have a look?

**Comment by moxious** [comment]
are you referring to something like this https://github.com/technosophos/common-chart

**Comment by gwvandesteeg** [comment]
Examples?  sure
Pretty much every chart here: https://github.com/helm/charts/tree/master/stable
Explicit references:
* https://github.com/helm/charts/blob/master/stable/chartmuseum/templates/_helpers.tpl
* https://github.com/helm/charts/blob/master/stable/cert-manager/templates/_helpers.tpl
* https://github.com/helm/charts/blob/master/stable/elasticsearch/templates/_helpers.tpl
* https://github.com/helm/charts/blob/master/stable/external-dns/templates/_helpers.tpl
* https://github.com/helm/charts/blob/master/stable/grafana/templates/_helpers.tpl
* https://github.com/helm/charts/blob/master/stable/jaeger-operator/templates/_helpers.tpl
* https://github.com/helm/charts/blob/master/stable/jenkins/templates/_helpers.tpl

**Comment by moxious** [comment]
@gwvandesteeg have a look at this PR and see if it does what you're after.   https://github.com/neo4j-contrib/neo4j-helm/pull/53

I had a look at some of the examples, and they're not very standard other than the use of a single fullNameOverride value.  Other than that, there's some real diversity in there in terms of supporting prefixes, suffixes, global prefix/suffix, and special naming rules.   I've opted for a simple prefix, suffix, and full name override.

For example:

```
helm install mygraph . --set acceptLicenseAgreement=yes --set neo4jPassword=mySecretPassword --set readReplica.numberOfServers=1 --set fullnameOverride=graph --set fullnamePrefix=pre
```

Will end up producing a statefulset called "pregraph".

**Comment by gwvandesteeg** [comment]
Perfect, PR looks good.  Does everything I'd like it to do and more (wasn't expecting the prefix/suffix material).

**Comment by moxious** [comment]
PR merged.  Closing this issue -- this fix will be included with the 4.1.0-3 release, which will get cut after I'm done with some other PRs in the queue right now.
